File: rides-service/src/main/java/com/modsen/dto/promo/PromoCodeRequest.java
package com.modsen.dto.promo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PromoCodeRequest {
    private String name;
    private LocalDate fromDate;
    private LocalDate endDate;
    private Integer discount;
}

File: promo-code-service/src/main/kotlin/com/modsen/promocodeservice/dto/PromoCodeRequest.kt
package com.modsen.promocodeservice.dto
import com.modsen.promocodeservice.constants.PromoCodeServiceConstant
import jakarta.validation.constraints.AssertTrue
import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import java.time.LocalDate

data class PromoCodeRequest(

    var name: String,
    var fromDate: LocalDate,
    var endDate: LocalDate,
    @Min(value = 1, message = PromoCodeServiceConstant.Validation.Message.NOT_RIGHT_RANGE)
    @Max(value = 100, message = PromoCodeServiceConstant.Validation.Message.NOT_RIGHT_RANGE)
    var discount: Int
) {
    @AssertTrue(message = PromoCodeServiceConstant.Validation.Message.NOT_RIGHT_DATE)
    fun isEndTimeAfterStartTime(): Boolean {
        return fromDate.isAfter(LocalDate.now()) && endDate.isAfter(fromDate)
    }
}